packages/expo-updates/cli/utils/args.ts

```diff
@@ -40,3 +40,14 @@ export function assertArgs(schema: arg.Spec, argv: string[]): arg.Result<arg.Spe
     throw error;
   }
 }
+
+export function assertArg(args: arg.Result<arg.Spec>, name: any, type: 'string' | 'number'): any {
+  const value = args[name];
+  if (value === undefined || value === null) {
+    Log.exit(`${name} must not be null`);
+  }
+  if (typeof value !== type) {
```

Comment:
Doesn't assertArgs already check the type?

Reply:
I think it coerces the type according to the readme: https://github.com/vercel/arg/blob/main/README.md
Though it doesn't support required args: https://github.com/vercel/arg/blob/main/README.md#how-do-i-require-an-argument-with-arg

I think I prefer to leave it just to be sure since it's unclear to me what the lib does and doesn't do.

Reply:
arg does assert the value i.e. `{ foobar: String }` + `--foobar` = assertion about a missing value.

Reply:
Trying this out:
- it coerces the type, so using Number constructor in `arc` but passing a string argument in the CLI will just result in the value being `NaN` rather than throwing. Don't see a fix for this since the coercion happens in the framework. Maybe all args should be String and then conversion to Number should happen in application code?
- it doesn't have a way to require args be present. Failing to pass an arg doesn't throw in `arg` code and instead makes to the application code.
- Passing empty for an arg also results in `NaN` I believe, so `--arg=` is also not validated by the framework.

Reply:
The answer to ["how do I require an arg?" is "do it yourself"](https://github.com/vercel/arg#how-do-i-require-an-argument-with-arg), so we need a function like this one.

It sounds like the type checking in this function isn't needed because assertArgs handles coercion. We may want to add a NaN check (Number.isNaN) in addition to the null/undefined checks.

Suggestion: rename this function to `requireArg` or `enforceArg` since what it does is different the existing `assertArgs` function and using a different verb will make that distinction clearer. Remove the type check since assertArgs does coercion. Maybe add a NaN check.

Reply:
Sounds good. Note that this whole discussion also applies to the new expo CLI from which these argument utils were copypasta'd.